app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,23 @@
+class FriendshipsController < ApplicationController
```

Comment:
Are we not using a safeguard here to ensure that only authenticated users can add or delete a friend? We can use perhaps a `before_action` callback here and have a private method that forces logging in if the user attempting to perform certain actions are not logged in

Reply:
We actually have an authenticate_user! in the application controller, so any action requires authentication.